lib/spack/spack/build_systems/cached_cmake.py

```diff
@@ -263,16 +263,10 @@ def initconfig_hardware_entries(self):
             cuda_flags = []
             archs = spec.variants["cuda_arch"].value
             if archs[0] != "none":
-                arch_str = ";".join(archs)
-                entries.append(
-                    cmake_cache_string("CMAKE_CUDA_ARCHITECTURES", "{0}".format(arch_str))
-                )
-                # TODO: Additional definitions that may not be needed anymore
-                # This is an imperfect merge of the radiuss way and the "spack"
-                # way: radiuss used to define only the first arch in the list,
-                # even for CMAKE_CUDA_ARCHITECTURE. What do we want?
-                cuda_flags.append("-arch sm_{0}".format(archs[0]))
-                entries.append(cmake_cache_string("CUDA_ARCH", "sm_{0}".format(archs[0])))
+                entries.append(cmake_cache_string("CMAKE_CUDA_ARCHITECTURES", ";".join(archs)))
+                # Additional definitions that may not be needed anymore
+                cuda_flags = cuda_flags(archs)
+                entries.append(cmake_cache_string("CUDA_ARCH", " ".join(cuda_flags)))
```

Comment:
I may be wrong, but I think `CMAKE_CUDA_ARCHITECTURES` covers it and we don't need `CUDA_ARCH` anymore.

Reply:
correct.. `CUDA_ARCH` was never a real variable and was only referenced in other variables that then got used, for example, `CMAKE_CUDA_FLAGS`. It can be removed.

Reply:
I want to believe you. I’ll try, but I’ll test in every RADIUSS projects. I seem to remember it wasn’t that simple.

Reply:
I think you're right about that for older versions of cmake, we had workarounds where they didn't quite always behave the same IIRC, something like before 3.18 or 3.21 maybe? If we're requiring what I think we are now though it should be fine. Especially with the updated BLT.

Reply:
Work fine. This is done.